src/libraries/System.Net.Security/src/System/Net/Security/SslStream.Protocol.cs

```diff
@@ -227,7 +229,7 @@ internal void CloseContext()
                     if (CertificateValidationPal.EnsureStoreOpened(isServer) is X509Store store)
                     {
                         X509Certificate2Collection certs = store.Certificates;
-                        X509Certificate2Collection found = certs.Find(X509FindType.FindByThumbprint, certHash, false);
+                        X509Certificate2Collection found = certs.FindByThumbprint(HashAlgorithmName.SHA512, certHash);
```

Comment:
Note that `FindByThumbprint` has different lifetime semantics than `Find`. `Find` would return an up-refed certificate, independent from the store. `FindByThumbprint` does not.

Since you iterate through all of the `certs` on line 261-264, `found` will get disposed.

Reply:
I am a bit worried that this did not cause any of the existing tests to fail.